src/feature_pipeline.cc

```diff
@@ -19,11 +21,19 @@ namespace alex_asr {
     {
         OnlineFeatureInterface *prev_feature;
 
-        KALDI_VLOG(3) << "Feature MFCC "
-                      << config.mfcc_opts.mel_opts.low_freq
-                      << " " << config.mfcc_opts.mel_opts.high_freq;
-        prev_feature = mfcc_ = new OnlineMfcc(config.mfcc_opts);
-        KALDI_VLOG(3) << "    -> dims: " << mfcc_->Dim();
+        if(config.feature_type == DecoderConfig::MFCC) {
+            KALDI_VLOG(3) << "Feature MFCC "
+                          << config.mfcc_opts.mel_opts.low_freq
+                          << " " << config.mfcc_opts.mel_opts.high_freq;
+            prev_feature = base_feature_ = new OnlineMfcc(config.mfcc_opts);
+            KALDI_VLOG(3) << "    -> dims: " << base_feature_->Dim();
+        } else if(config.feature_type == DecoderConfig::FBANK) {
+            KALDI_VLOG(3) << "Feature FBANK "
+                          << config.fbank_opts.mel_opts.low_freq
+                          << " " << config.fbank_opts.mel_opts.high_freq;
+            prev_feature = base_feature_ = new OnlineFbank(config.fbank_opts);
+            KALDI_VLOG(3) << "    -> dims: " << base_feature_->Dim();
+        }
```

Comment:
I know that this state cannot happen now, but I'd add welcome an "else {" block to be future-proof.